_posts/2023-02-19-a-lean-dataops-platform.md

```diff
@@ -0,0 +1,158 @@
+
+For Mechanical Rock, we wanted to build, using data-ops practices, an internal data platform that facilitates a few key objectives. 
+
+These objectives are the ability to aggregate data from the different SaaS products we use, creating dashboards for some high level metrics, trial a few new technologies and facilitate some level of data analysis. The first three use cases, by some measures, are marked as complete today and the current state of the platform is discussed in this blog.
+
+The good and proper practices of "data-ops" or any other buzz-word to describe good engineering practices include our CI/CD pipelines, governance checks and testing as well as data infrastructure as code. 
+
+To create our data platform we leveraged multiple greenfield technologies in order to decrease our time to value as well as minimizing the maintenance cost of the platform. All the while ensuring our dev-ops dream of good engineering practice lives on in the data space.
+
+# Infrastructure
+## Designing The Data Platform 
+In order to build our data platform we needed to load data from our sources and transform it into a usable structure while also ensuring the security of the platform, and thus the data, was up to scratch. To rapidly create this platform while not cutting the corners of these intentions we decided on using the SaaS technologies Snowflake and Fivetran for our storage and data loading respectively. The resources needed could be created both quickly, and more importantly in code, using the Terraform provider for each of these technologies. This, coupled with strong governance, meant our platform and data are secure. However, these tools are not without their drawbacks, and these will be discussed later on. In addition, we are also using DBT for data transformation, which of course is also codified. 
+
+![Flow of Data](../img/data-driven-platform-images/architecture_diagram_1.jpg)
+
+The orchestration of Fivetran and DBT is done by Dagster to ensure the dependencies between the two tools is robust. 
+
+![orchestration](../img/data-driven-platform-images/architecture_diagram_3.jpg)
+
+Our CI/CD pipeline was created with Github actions, the checks for which ensure that governance policies are maintained and Terraform is (hopefully) working as expected. For our deployments it ensures that the infrastructure is provisioned in a consistent and reliable way with the correct version control as well as change and approval processes. 
```

Comment:
Github actions
should be 
Github Actions

Reply:
yes